portfolio/src/main/java/com/google/sps/servlets/FormServlet.java

```diff
@@ -0,0 +1,75 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.ArrayList;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that returns user comments. */
+@WebServlet("/handle-comment")
+public class FormServlet extends HttpServlet {
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Retrieve newest user comment and store it as a Comment Entity
+    String userComment = request.getParameter("user-comment");
+    Entity commentEntity = new Entity("Comment");
+    commentEntity.setProperty("text", userComment);
+
+    // Store the Comment Entity in Datastore 
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(commentEntity);
+
+    // Redirect back to the HTML page
+    response.sendRedirect("/comment.html");
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query query = new Query("Comment");
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
```

Comment:
Should this datastore be a private property on the class? Or is it important that it gets created at every get and post?

Reply:
I originally based my implementation off of the tutorial's example where they created it at every get and post, but I made it private and the functionality was not affected & I could not find any justification for needing it at every get and post within the tutorial instructions. I just pushed a version where datastore is private--please let me know if you meant something different by private property.

Reply:
It depends on how expensive it is to build, and if there is any configuration baked into it. A typical database connection (like MySQL) would actually be managed by the app framework context, it's a global concern. The app framework manages a pool of connections, and injects a connection into the servlet when it receives a request.

Though, MySQL connections need to be maintained, and are much more expensive. I don't know much of how DataStore works, but it looks to me to be a stateless, HTTP based API. So the connection overhead is basically non-existent.